### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW47 2024)

### PR DESCRIPTION
**Pull request by iclsrc**
LLVM: llvm/llvm-project@f8bae3af74e7c60d996f0d331cad04f2eace7f8f
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@9207ef2aa150773

### CODE REVIEW

**Comment by jsji** [comment]
Something wrong with github. I don't see any conflicts locally with git.

**Comment by jsji** [comment]
This is ready for review.
- [ ] [Update sycl test using new splat syntax](https://github.com/intel/llvm/pull/16165/commits/6387a057a60f063dd1dd8944b67c969898db7e85) @intel/dpcpp-kernel-fusion-reviewers @intel/unified-runtime-reviewers 
- [ ] [Update enqueue_kernel.cl after 88b18319c943](https://github.com/intel/llvm/pull/16165/commits/6c524eb7a0a32b46a10bb11114820e4240d2275d) @intel/dpcpp-spirv-reviewers 
- [ ] [libclc: fix clcmaro.h path for rsqrt.cl](https://github.com/intel/llvm/pull/16165/commits/69da5c5689f8e42da3b136f2bbf3422cdeadbdf7) @frasercrmck @intel/llvm-reviewers-cuda 
- [ ] [[NVPTX] Update test w/ nvvm.ldg.global.* removal in](https://github.com/intel/llvm/pull/16165/commits/16e0fe56e8702c9cc252dd5637c2bfd33668a858) https://github.com/intel/llvm/commit/fb33af08e4c105a05855f8beeb972d493410e72f @JackAKirk @intel/llvm-reviewers-cuda

**Review by frasercrmck — APPROVED** [review]
libclc LGTM

**Comment by jsji** [comment]
Ping @intel/llvm-reviewers-runtime @intel/dpcpp-spirv-reviewers @intel/llvm-reviewers-cuda

File: llvm-spirv/test/transcoding/enqueue_kernel.cl [hunk 1]
```diff
@@ -17,11 +17,11 @@
 // CHECK-SPIRV: EntryPoint {{[0-9]+}} [[BlockKer5:[0-9]+]] "__device_side_enqueue_block_invoke_5_kernel"
 // CHECK-SPIRV: Name [[BlockGlb1:[0-9]+]] "__block_literal_global"
 // CHECK-SPIRV: Name [[BlockGlb2:[0-9]+]] "__block_literal_global.1"
-// CHECK-SPIRV: Name [[#InvokeFunc1:]] "__device_side_enqueue_block_invoke_kernel"
-// CHECK-SPIRV: Name [[#InvokeFunc2:]] "__device_side_enqueue_block_invoke_2_kernel"
-// CHECK-SPIRV: Name [[#InvokeFunc3:]] "__device_side_enqueue_block_invoke_3_kernel"
-// CHECK-SPIRV: Name [[#InvokeFunc4:]] "__device_side_enqueue_block_invoke_4_kernel"
-// CHECK-SPIRV: Name [[#InvokeFunc5:]] "__device_side_enqueue_block_invoke_5_kernel"
+// CHECK-SPIRV: Name [[#InvokeFunc1:]] "__device_side_enqueue_block_invoke"
```
Comment (MrSidims) [inline llvm-spirv/test/transcoding/enqueue_kernel.cl]: It's different from https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/test/transcoding/enqueue_kernel.cl . I'm ok to merge it as is, I've created internal tracker to investigate this

**Review by uditagarwal97 — APPROVED** [review]
Changes in `sycl/test/check_device_code/vector/vector_math_ops.cpp` due to use of `splat` LGTM.

**Comment by jsji** [comment]
@intel/llvm-gatekeepers I think this is ready for merge. Thanks.

File: llvm-spirv/test/transcoding/enqueue_kernel.cl (hunk 1 above)
Comment (jsji) [reply llvm-spirv/test/transcoding/enqueue_kernel.cl]: Thanks @MrSidims

**Comment by sarnex** [comment]
Is the command wrong or is the bot broken?

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Wed 04 Dec 2024 01:18:19 AM UTC --- Merge failed with error: PR is not clean for merge. Please examine ci check status before merge.

**Comment by DoyleLi** [comment]
> > Is the command wrong or is the bot broken?
> 
> Looks like the bot is failing again due to old git version. @DoyleLi

Hi, The issue is resolved. Will take effect in next pulldown.

**Pull request merged**